proposals/csharp-7.0/tuples.md

```diff
@@ -399,39 +451,67 @@ static class Extensions
 
 ```csharp
 void Deconstruct(out byte x, out byte y) ...;
 ```
 
-The compiler translates
+the compiler translates
 
 ```csharp
 (int x, int y) = p;
 ```
 
-equivalently to:
+to:
 
 ```csharp
 p.Deconstruct(out byte __x, out byte __y);
 (int x, int y) = (__x, __y);
 ```
 
+The evaluation order of deconstruction assignment expressions is "breadth first":
+
+**ISSUE:** This is the first place I've seen deconstruction mentioned in the same breath as assignment. If deconstruction can only occur on the LHS of a (simple?, compound?) assignment, should deconstruction be covered under assignment rather than in its own section? If the two are in spearate sections, the expression grammar needs to have the new production, tuple_deconstruction_expression, plugged into it. From my experience with tuples, patterns, and _ wildcards in other languages, I'm thinking that the V7.0 support for tuple deconstruction is the beginning of a more general pattern support mechanism to come later. If that is the case, perhaps we should admit that and start using pattern-related terminology and organization. If so, that suggests tuple deconstruction should be separate from the assignment operator, in a (pre-)patterns section that will be expanded over future spec versions.
+
+1. Evaluate the LHS: Evaluate each of the expressions inside of it one by one, left to right, to yield side effects and establish a storage location for each.
+1. Evaluate the RHS: Evaluate each of the expressions inside of it one by one, left to right to yield side effects
+1. Convert each of the RHS expressions to the LHS types expected, one by one, left to right.
+1. Assign each of the conversion results from Step 3 to the storage locations found in (???)
+
+> **Note to reviewers**: I found this in the LDM notes for July 13-16, 2016. I don't think it is still accurate:
+
+**ISSUE:** flesh out the following example
+
+\[Example:
+```csharp
+string x;
+byte y;
+
+(x, y) = (y, x); // swap!
+```
+end example\]
+
+A deconstructing assignment is a *statement-expression* whose type could be `void`.
+
 ### Additions to [Classes](../../spec/classes.md)
 
-> The following note should be added to the end of the section on [extension methods](../../spec/classes.md#extension-methods):
+### Extension methods
+
+> Add the following note to the end of the section on [extension methods](../../spec/classes.md#extension-methods):
 
-> note: 
->Extension methods on a tuple type apply to tuples with different element names:
+[*Note*: Extension methods on a tuple type apply to tuples with different element names:
 
 ```csharp
 static void M(this (int x, int y) t) { ... }
 
 (int a, int b) t = ...;
 t.M(); // Sure
 ```
+*endnote*].
 
-> endnote.
\ No newline at end of file
+**ISSUE:** Explain what the comment "Sure" means.
```

Comment:
This should be saying that the extension method `M` is a candidate method, even though the tuple `t` has different element names (`a`, ` b`) than then formal parameter of M (`x`, `y`).

Reply:
OK; I added that explanation to the end of the Note.